ddtrace/tracer/transport.go

```diff
@@ -183,6 +186,24 @@ func (t *httpTransport) endpoint() string {
 	return t.traceURL
 }
 
+// getAgentURL determines the trace agent URL. The value held by environment variable
+// DD_TRACE_AGENT_URL takes precedence over other configuration options.
+func getAgentURL(addr string) string {
+	if agentURL := os.Getenv("DD_TRACE_AGENT_URL"); agentURL != "" {
```

Comment:
```suggestion
	if agentURL := os.Getenv("DD_TRACE_AGENT_URL"); agentURL == "" {
	    return resolveAgentAddr(addr)
	}
```
Try to stick to the left on the "happy path".